@types/algoliasearch-helper: Add missing method getQueryParams to SearchParameters

Kamshak opened the pull request:
Adds typings for method getQueryParams() from https://github.com/algolia/algoliasearch-helper-js/blob/develop/src/SearchParameters/index.js#L1534

This is useful for example for using searchOnce and using the SearchParameters API:
```ts
        const originalState = helperWithSearchRequest.getState();
        // SearchParameters is immutable - this does not modify the original search
        const sameSearchButForTrips: SearchParameters = originalState.addFacetRefinement('type', 'trip');
        return helper.searchOnce(sameSearchButForTrips.getQueryParams());
``` 
(if any of the original maintainers read this, it would also be nice to have a 'searchOnce' method that accepts a SearchParameters object)
#### Template Info

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/algolia/algoliasearch-helper-js/blob/develop/src/SearchParameters/index.js#L1534
- [x] Increase the version number in the header if appropriate: Not needed, present in current version
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

PikachuEXE commented:
Why not also update the test though?
Also it seems `getQueryParams` always return `object` instead of `any`

Haroenv commented on types/algoliasearch-helper/index.d.ts:
```diff
@@ -478,6 +478,7 @@ declare namespace algoliasearchHelper {
     getHierarchicalRefinement(facetName: string): string[];
     getNumericRefinements(facetName: string): SearchParameters.OperatorList[];
     getNumericRefinement(attribute: string, operator: SearchParameters.Operator): Array<number | number[]>;
+    getQueryParams(): any;
```
```suggestion
    getQueryParams(): SearchParameters;
```

Kamshak replied:
@Haroenv Would it make sense to also update helper.searchOnce to accept type QueryParameters?

Haroenv replied:
@Kamshak, yes, that's correct!

Kamshak commented:
@Haroenv 
I checked the code again, it looks like the query params that are returned by getQueryParams are restricted to "managedParameters" which is a subset of the full QueryParameters [1].

I've updated the types to reflect this. I've introduced a new (non-exported) type `ManagedParameters` which contains these types and is mapped from the `managedTypes` property (for which I also added typings). The field types map to the corresponding type in `SearchParameters` (and therefore also `QueryParameters` since `class SearchParameters implements QueryParameters`, but I felt this closer expresses what's actually happening).

I've added a test for the function and to make sure that the return can be used for querying

[1] https://github.com/algolia/algoliasearch-helper-js/blob/develop/src/SearchParameters/index.js#L1528

Haroenv commented:
that's a very good catch, thanks!
